packages/flutter_test/lib/src/controller.dart

```diff
@@ -143,7 +143,7 @@ class WidgetController {
   ///
   /// * Use [firstState] if you expect to match several states but only want the first.
   /// * Use [stateList] if you expect to match several states and want all of them.
-  T state<T extends State>(Finder finder) {
+  T state<T extends State>(Finder finder) { // ignore: strong_mode_not_instantiated_bound
```

Comment:
State is marked `@optionalTypeArgs`. I would hope it would default to the broadest type possible (dynamic, or, in the case of a `<T extends Q>`, Q).